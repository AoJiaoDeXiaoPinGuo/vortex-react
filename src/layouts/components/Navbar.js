import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { NavLink as Link } from 'react-router-dom'
import { Collapse, Navbar, NavbarToggler, NavbarBrand, Nav, NavLink } from 'reactstrap'

const location = {
  pathname: '/route/8080',
  state: { fromDashboard: true },
  query: {
    a: 2
  }
}

export default class CustomNavbar extends Component {
  static propTypes = {
    isAuthenticated: PropTypes.bool.isRequired,
    signOutWithCb: PropTypes.func.isRequired,
    history: PropTypes.object.isRequired,
  }

  state = {
    isOpen: false
  }

  toggle = () => {
    this.setState({
      isOpen: !this.state.isOpen
    })
  }

  signOut = (e) => {
    this.props.signOutWithCb(() => this.props.history.push('/'))
  }

  render() {
    return (
      <Navbar color="faded" light toggleable>
        <NavbarToggler right onClick={this.toggle} />
        <NavbarBrand exact to="/" tag={Link}>Vortex React</NavbarBrand>
        <Collapse isOpen={this.state.isOpen} navbar>
          <Nav className="ml-auto" navbar>
            <NavLink to="/counter" activeClassName="active" tag={Link}>Counter</NavLink>
            <NavLink to="/zenPage" activeClassName="active" tag={Link}>Zen</NavLink>
            <NavLink to="/elapse" activeClassName="active" tag={Link}>Elapse</NavLink>
            <NavLink to={location} activeClassName="active" tag={Link}>Route</NavLink>
            <NavLink to="/pageNotFound" activeClassName="active" tag={Link}>404</NavLink>
            <NavLink href="https://github.com/YutHelloWorld/vortex-react">Github</NavLink>
            {
              this.props.isAuthenticated ? <NavLink href="javascript:;" onClick={this.signOut}>Sign out</NavLink>
                : <NavLink to="/login" tag={Link}>Sign in</NavLink>
            }
          </Nav>
        </Collapse>
      </Navbar>
    )
  }
}
