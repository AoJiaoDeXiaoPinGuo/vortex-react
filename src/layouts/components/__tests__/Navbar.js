import React from 'react';
import ReactDOM from 'react-dom';
import { HashRouter } from 'react-router-dom';
import Navbar from '../Navbar';

it('(HoC) AsyncLoad', () => {
  const div = document.createElement('div');
  ReactDOM.render(
    <HashRouter>
      <Navbar signOut={() => {}} />
    </HashRouter>,
    div
  );
});
