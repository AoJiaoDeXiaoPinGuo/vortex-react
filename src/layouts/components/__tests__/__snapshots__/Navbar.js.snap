// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Layout) Navbar 1`] = `
<nav
  className="navbar navbar-toggleable navbar-light bg-faded"
>
  <button
    className="navbar-toggler navbar-toggler-right"
    onClick={[Function]}
    type="button"
  >
    <span
      className="navbar-toggler-icon"
    />
  </button>
  <a
    aria-current="true"
    className="navbar-brand active"
    href="#/"
    onClick={[Function]}
    style={Object {}}
  >
    Vortex React
  </a>
  <div
    className="collapse navbar-collapse"
    style={Object {}}
  >
    <ul
      className="ml-auto navbar-nav"
    >
      <a
        aria-current={false}
        className="nav-link"
        href="#/zenPage"
        onClick={[Function]}
        onMouseOver={[Function]}
        style={undefined}
      >
        Zen
      </a>
      <a
        className="nav-link"
        href="#"
        onClick={[Function]}
      >
        Sign out
      </a>
      <a
        className="nav-link"
        href="https://github.com/YutHelloWorld/vortex-react"
        onClick={[Function]}
      >
        Github
      </a>
    </ul>
  </div>
</nav>
`;

exports[`(Layout) Navbar 2`] = `
<nav
  className="navbar navbar-toggleable navbar-light bg-faded"
>
  <button
    className="navbar-toggler navbar-toggler-right"
    onClick={[Function]}
    type="button"
  >
    <span
      className="navbar-toggler-icon"
    />
  </button>
  <a
    aria-current="true"
    className="navbar-brand active"
    href="#/"
    onClick={[Function]}
    style={Object {}}
  >
    Vortex React
  </a>
  <div
    className="collapse navbar-collapse"
    style={Object {}}
  >
    <ul
      className="ml-auto navbar-nav"
    >
      <a
        aria-current={false}
        className="nav-link"
        href="#/zenPage"
        onClick={[Function]}
        onMouseOver={[Function]}
        style={undefined}
      >
        Zen
      </a>
      <a
        className="nav-link"
        href="#"
        onClick={[Function]}
      >
        Sign out
      </a>
      <a
        className="nav-link"
        href="https://github.com/YutHelloWorld/vortex-react"
        onClick={[Function]}
      >
        Github
      </a>
    </ul>
  </div>
</nav>
`;
